osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -112,29 +113,57 @@ public WorkingBeatmap CreateNew(RulesetInfo ruleset, APIUser user)
         /// The new difficulty will be backed by a <see cref="BeatmapInfo"/> model
         /// and represented by the returned <see cref="WorkingBeatmap"/>.
         /// </summary>
-        public virtual WorkingBeatmap CreateNewBlankDifficulty(BeatmapSetInfo beatmapSetInfo, RulesetInfo rulesetInfo)
+        public virtual WorkingBeatmap CreateNewDifficulty(NewDifficultyCreationParameters creationParameters)
```

Comment:
Thoughts on putting the `CreateBlank` parameter as a method parameter instead of in the parameters class? Feels like something which would better exist at a top level and be a forced decision. I'd probably call it `copyExistingHitObjects` as well.

Also wondering if the remaining reference objects could better fit inside `EditorState` itself..

Reply:
I'll try hoisting `CreateBlank` out.

As for the other objects - `EditorState` is also used in more contexts, such as the gameplay test screen, so not sure putting the other properties there is going to make sense in all contexts.

Reply:
I've ended up removing `NewDifficultyCreationParameters` entirely after some restructuring.